### DISCUSSION TITLE
How to do a scheduled logical backup to azure blob storage

### DISCUSSION THREAD

**Discussion by adminnz** [cloudnative-pg/cloudnative-pg#2116] [Q&A]
From my understanding barman performs physical backups and that is what gets configured when you define a ScheduledBackup with the cluster having the backup configuration set.

This works fine, however our postgres cluster has multiple databases within it (i.e. not 1 db per cluster), and would like to know what would be the way to have a scheduled logical backup of each database within the cluster to azure blob storage.

Is this possible?

e.g.
DatabaseA -> backed up to Azure Storage container in a path like /DatabaseA/
DatabaseB -> backed up to Azure Storage container in a path like /DatabaseB/

**Comment by itay-grudev** [comment]
The Barman backups are not at a specific database level. Barman backups are physical backups at the filesystem level that include all databases. In addition Barman stores WAL so you can have Point-in-time recovery (PITR).

So to answer your question, no you cannot create a Scheduled Backup of individual databases inside your cluster. 

If you need to do that you're going to need another mechanism - like `pg_dump`, but those would not support PITR increasing your RPO.

**Comment by gbartolini** [comment]
Scheduled logical backups are not a priority at the moment from our point of view, given our capacity and also our pursue of business continuity. Logical backups, as @itay-grudev says, are snapshots and are not suitable for RPO=0 backups like the ones we are currently targeting. We'd like to improve our posture in that regard first with, for example, support for Kubernetes Volume Snapshot API.

Nothing prevents you from running jobs that run `pg_dump` (as shown in https://cloudnative-pg.io/documentation/current/troubleshooting/#emergency-backup) and relay them on S3. Obviously, if we had an API for that, similar to the ScheduleBackup resource but for logical dumps, that would be better - but the workaround is there.

In any case, if somebody wants to work on that, we'd be happy to review it. After all, we already have most of the primitives in the code - as the "import" facility relies on pg_dump.

**Comment by sxd** [comment]
I'm turning this into a discussion since it looks more like a question/idea than an issue now

Feel free to open an issue if something comes up from this discussion

**Comment by JakobStadlhuber** [comment]
Has anyone already contributed regardin the ScheduleBackup for logical dumps?

**Comment by jcpunk** [comment]
I'll admit interest in this sort of feature.

Logical backups aren't perfect, but they provide a level of comfort for our auditors and disaster recovery workflows.  Having a nice built in way of taking a logical backup at a scheduled interval, expiring old copies, and writing it to a PVC would be very handy.
